src/exo/stdlib/scheduling.py

```diff
@@ -94,10 +94,13 @@
     autolift_alloc,
 )
 
+
 from .analysis import (
     check_call_mem_types,
 )
 
+from exo.range_analysis import IndexRange, bounds_inference, get_affected_idxs
+
```

Comment:
We need to be careful about illegally importing compiler code in the standard library.

Perhpas these functions should have an API wrapper similar to how we have an API wrapper for scheduling ops in API_scheduling.py. We should also make sure to maintain a similar calling/naming convention. I am not sure what's a good name for it? API_introsepction.py, API_analysis.py?

Reply:
I will fix this in a future PR to better separate user-level and compiler-level bounds inference.